src/test/java/com/aws/greengrass/logmanager/model/LogFileTest.java

```diff
@@ -97,6 +97,59 @@ void GIVEN_log_file_with_more_than_target_lines_but_two_lines_WHEN_calculate_fil
         writeFile(file, builder.toString().getBytes(StandardCharsets.UTF_8));
         String fileHash = file.hashString();
         assertEquals(fileHash, expectedHash);
-        file.delete();
+    }
+
+    @Test
+    void GIVEN_logFileTrackingHardlink_WHEN_trackedFileRotates_THEN_itDeletesTheOriginalFile() throws IOException {
+        // Given
+
+        Path testPath = directoryPath.resolve("itDeletesTheOriginalFile");
+        Utils.createPaths(testPath.resolve("hardlinks"));
+
+        File file = TestUtils.createFileWithContent(
+                testPath.resolve("test.log"), "rotated");
+        LogFile logFile = LogFile.of(file, testPath.resolve("hardlinks"));
+
+        // When
+
+        rotateFilesByRenamingThem(new File[]{file});
+        logFile.delete();
+
+        // Assert
+
+        File[] directoryFiles = Arrays.stream(testPath.toFile().listFiles())
+                        .filter(File::isFile)
+                        .toArray(File[]::new);
+        assertEquals(directoryFiles.length, 1);
+
+        File remainingFile = directoryFiles[0];
+        assertEquals(remainingFile.toPath(), file.toPath());
+    }
+
+    @Test
+    void GIVEN_logFileTrackingRegularFile_WHEN_trackedFileRotates_THEN_itDeletesTheWrongFile() throws IOException {
```

Comment:
Nit: Add a comment her that this happens in older versions of LM or if the hardlinks are not used.